### DISCUSSION TITLE
Help with Waveshare RP2040-LCD-1.28 w/GC9A01A driver

### DISCUSSION THREAD

**Discussion by RickJHarris** [Bodmer/TFT_eSPI#2025] [Raspberry Pi Pico]
I'm attempting to get TFT_eSPI working on this Waveshare Pico/1.28" with a GC9A01A driver combo - https://www.waveshare.com/wiki/RP2040-LCD-1.28 , but get only a blank screen, and have run out of ideas. It works with Adafruit GFX, so I know my pin assignments are correct. 

Here's my User_Setup.h
#define RPI_DISPLAY_TYPE // 20MHz maximum SPI
#define GC9A01_DRIVER
#define TFT_WIDTH  240 // ST7789 240 x 240 and 240 x 320
#define TFT_HEIGHT 240 // ST7789 240 x 240
#define TFT_MISO -1 
#define TFT_MOSI 11 // In some display driver board, it might be written as "SDA" and so on.
#define TFT_SCLK 10
#define TFT_CS   9 //22  // Chip select control pin
#define TFT_DC   8  //5  // Data Command control pin
#define TFT_RST  12  // Reset pin (could connect to Arduino RESET pin)
#define TFT_BL   25  // LED back-light
#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define SMOOTH_FONT
#define TFT_SPI_PORT 1 // Set to 0 if SPI0 pins are used, or 1 if spi1 pins used
#define SPI_FREQUENCY  20000000

Here's my Read_User_Setup output (Including SPI Port#)
[code]
TFT_eSPI ver = 2.4.72
Processor    = RP2040
Transactions = Yes
Interface    = SPI
SPI Port = SPI1
Display driver = 9A01
Display width  = 240
Display height = 240

MOSI    = GPIO 11
SCK     = GPIO 10
TFT_CS   = GPIO 9
TFT_DC   = GPIO 8
TFT_RST  = GPIO 12

TFT_BL   = GPIO 25

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Smooth font enabled

Display SPI frequency = 20.00
[/code]

And the Arduino IDE verbose output when compiling Color_Test is in the attached file.

[verbose_ide.txt](https://github.com/Bodmer/TFT_eSPI/files/9574910/verbose_ide.txt)

Any help would be appreciated. Thanks.

**Comment by RickJHarris** [comment]
I contacted Waveshare and their tech support provided me with a working User_Setup.h . I was missing the line #define TFT_BACKLIGHT_ON HIGH 

The RP2040-LCD-1.28 is a hell of a deal. Only a couple dollars more than a standalone 1.28 round LCD, and it includes an integrated Pi Pico.

Working User_Setup.h attached. 
[User_Setup.zip](https://github.com/Bodmer/TFT_eSPI/files/9601198/User_Setup.zip)

**Comment by lukasreinfurt** [comment]
For the Waveshare RP2040-Touch-LCD-1.28 the setup is slightly different.
Here is my platformio.ini file that worked.
Note: For this board, `TFT_RST` is on pin 13 instead of 12, as shown above.

```
; PlatformIO Project Configuration File
;
;   Build options: build flags, source filter
;   Upload options: custom upload port, speed and extra flags
;   Library options: dependencies, extra library storages
;   Advanced options: extra scripting
;
; Please visit documentation for the other options and examples
; https://docs.platformio.org/page/projectconf.html

[env:waveshare_rp2040_lcd_1_28]
platform = raspberrypi
board = waveshare_rp2040_lcd_1_28
framework = arduino
lib_deps =
	bodmer/TFT_eSPI@^2.5.33
build_flags =
  -Os
  -DUSER_SETUP_LOADED=1

  -DGC9A01_DRIVER=1
  -DTFT_SPI_PORT=1
  -DTFT_WIDTH=240
  -DTFT_HEIGHT=240
  -DTFT_MOSI=11
  -DTFT_SCLK=10
  -DTFT_CS=9
  -DTFT_DC=8
  -DTFT_RST=13
  -DTFT_BL=25
  -DTFT_BACKLIGHT_ON=HIGH
  
  -DLOAD_GLCD=1   # Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
  -DLOAD_FONT2=1  # Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
  -DLOAD_FONT4=1  # Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
  -DLOAD_FONT6=1  # Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
  -DLOAD_FONT7=1  # Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
  -DLOAD_FONT8=1  # Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
  -DLOAD_GFXFF=1  # FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
  
  -DSMOOTH_FONT=1
  
  -DSPI_FREQUENCY=27000000
  -DSPI_READ_FREQUENCY=5000000
```
